parcels/field.py

```diff
@@ -205,6 +205,38 @@ def get_dim_filenames(cls, filenames, dim):
         else:
             return filenames
 
+    @classmethod
```

Comment:
well, the question here really is: do we need any class-internal or object-internal functions ? having this be a class method is already MUCH nicer than some function-in-function construct, but the way I see it, it could also be a static method. @erikvansebille it's up to you - you can leave it this way, which is fine, or you can make it a static method. Either way you choose, I approve it.